src/plugins/main.js

```diff
@@ -119,9 +119,17 @@ export default async (context) => {
     }
   }
 
+  const redirectToNewLocaleRoute = newLocale => {
```

Comment:
Hmm...
I've refactored this into separate function to call it also from `!initialSetup && app.i18n.differentDomains` to match old code. Currently the redirect will not be called from middleware for `differentDomains`.

But I have suspiction that current code is actually correct because it's not possible to redirect to different domain using that code (it uses `localePath` which doesn't consider different domains -- for some reason...). So I think that code has actually caused infinite redirect bug in old code.

Reply:
But then... It won't be able to redirect to different domain but I think it can work when route has overriden path through component/page options...

I guess I need to test this.
So many variables...

Reply:
Actually checked the concerning part and I think it's OK right now.

Beheavior differs from old code in this situation:
 - differentDomains = true
 - detectBrowserLanguage != false

Previously redirect was called, now it isn't. I think that's OK because that redirect can't redirect to different domain anyway and also it's based on browser language so that's not really compatible with having different domains which should itself dictate which language to choose.

Possibilly `detectBrowserLanguage` should even be force-disabled when `differentDomains` is set. Or alternatively, those could work together but code would need to be fixed to be able to redirect to different domain.

Reply:
And I've verified now that it actually fixes #411 without having to explicitly disable `detectBrowserLanguage`.